README.md

```diff
@@ -9,89 +10,119 @@ Allow for Registry credentials to be refreshed inside your Kubernetes cluster vi
-     "ecr:DescribeRepositories",
-     "ecr:ListImages",
-     "ecr:BatchGetImage"
-   ],
-   "Resource": "*"
-  }
-  ```
+1. Clone the repo and navigate to directory
 
-2b. If you are not running in AWS Cloud, then you can still use this tool! Edit & create the sample [secret](k8s/secret.yaml) and update values for AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, and AWS Account Id (base64 encoded)
+2. Input your `application_default_credentials.json` information into the `secret.yaml` template located [here](k8s/secret.yaml#L17):
+The value for `application_default_credentials.json` can be obtained with the following command:
 
-```bash
-echo -n "secret-key" | base64
+   ```bash
+   base64 -w 0 $HOME/.config/gcloud/application_default_credentials.json
+   ```
 
-kubectl create -f k8s/secret.yaml
-```
+3. Create the secret in kubernetes
 
-3. Create the replication controller. NOTE: If running on prem, no need to provide AWS_ACCESS_KEY_ID or AWS_SECRET_ACCESS_KEY since that will come from the EC2 instance.
+   ```bash
+   kubectl create -f k8s/secret.yml
+   ```
 
-  ```bash
-  kubectl create -f k8s/replicationController.yaml
-  ```
-4. Use awsecr-cred for name of imagePullSecrets on your deployment.yaml file.
+4. Create the replication controller:
 
-## How to setup running in GCR
+   ```bash
+   kubectl create -f k8s/replicationController.yaml
+   ```
+   
+## How to setup running in Docker Private Registry
 
 1. Clone the repo and navigate to directory
 
-2. Input your application_default_credentials.json information into the secret.yaml template located [here](k8s/secret.yaml#L17):
-The value for application_default_credentials.json can be obtained with the following command:
-```bash
-base64 -w $HOME/.config/gcloud/application_default_credentials.json
-```
+2. Edit the sample [secret](k8s/secret.yaml) and update values for `DOCKER_PRIVATE_REGISTRY_SERVER`, `DOCKER_PRIVATE_REGISTRY_USER`, and `DOCKER_PRIVATE_REGISTRY_PASSWORD` (base64 encoded).
+
+   ```bash
+   echo -n "secret-key" | base64
+   ```
 
 3. Create the secret in kubernetes
-```bash
-kubectl create -f k8s/secret.yml
-```
 
-3. Create the replication controller:
+   ```bash
+   kubectl create -f k8s/secret.yml
+   ```
+
+4. Create the replication controller:
 
-```bash
-kubectl create -f k8s/replicationController.yaml
-```
+   ```bash
+   kubectl create -f k8s/replicationController.yaml
+   ```
 
 ## DockerHub Image
 
-- https://hub.docker.com/r/upmcenterprises/awsecr-creds/
+- [upmcenterprises/registry-creds](https://hub.docker.com/r/upmcenterprises/registry-creds/)
 
 ## Developing Locally
 
 If you want to hack on this project:
 
 1. Clone the repo
-2. Build: `make binary`
+2. Build: `make build`
 3. Test: `make test`
-4. Run on your machine: ` go run ./main.go --kubecfg-file=<pathToKubecfgFile> --use-kubernetes-cluster-service=false
+4. Run on your machine: `./bin/registry-creds --kubecfg-file=<pathToKubecfgFile>` or `go run ./main.go --kubecfg-file=<pathToKubecfgFile>`
```

Comment:
I don't like the idea of having to build it first, then run. I like the workflow to just `go run`, see the output, stop it, then go run again. Do you have thoughts as to why this is better? I ends up being two steps each time you want to build.